### PR TITLE
add support for sequel

### PR DESCRIPTION
**Pull request by aiomaster**
I want to use this gem with sequel and don't have ActiveRecord in my project.
In sequel you don't have to hack around to share your connections - just add single_threded: true to your connection specification.
But I want to use the ajax magic of this gem, so I make this pull request to make it possible to use sequel and this gem.
What do you think?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,12 +138,16 @@ after :each do
 end
 ```
 
+## Sequel ##
+If you want to have shared database connections with sequel just
+add the option `single_threaded: true` to your sequel connection in test.
```
Comment (iangreenleaf) [inline README.md:143]: Would it be possible to set this automatically for people? It seems safe to assume that they want it single threaded any time this gem is active.

**Review by iangreenleaf — CHANGES_REQUESTED** [review]
This looks great and I would love to add Sequel support to the gem! Thanks for putting this together!

The one thing this definitely needs to be merge-ready is some tests. I may have to make some changes to the spec setup for it to be able to handle multiple DB adapters…

**Comment by iangreenleaf** [comment]
Cool, this is pretty much ready! Just let me know if it's possible to set the single-threaded option automatically.

**Comment by aiomaster** [comment]
Hi @iangreenleaf,
first thanks for your support at this! This is really awesome and motivating! :)

Maybe we should add tests for the connection sharing part.
As far as I see there are no such tests (I can comment out the code in shared_connection/active_record.rb completely and the tests will pass.
If we have such tests we could improve the sequel stuff.
I don't really know if it is possible to set the single_threaded mode after a connection was established.
But maybe yes (or we could close the current connection and reopen one with single_threaded: true).
At least we should warn in logs, that it would be better to set this option in the default test database connection settings. To raise an exception is maybe a bit too heavy, I think.

One last thing: In the docs you say "Right now this gem automatically fixes the following things:",
but for sequel there is not really something to "fix", cause there is this option.
Maybe we should say: "Right now this gem automatically handles the following things:".

So any idea, how we could test connection_sharing?

**Comment by aiomaster** [comment]
I asked @jeremyevans about enabling the single_threaded option on the fly.
Maybe he has a look and knows better than me.

**Comment by aiomaster** [comment]
Jeremy says:

> There isn't a supported method to do so.  You can try this:
> 
>   DB.disconnect
>   DB.instance_variable_set(:@pool,
>     Sequel::ConnectionPool.get_pool(DB,
>       DB.opts.merge(:single_threaded=>true)))
> 
> But it is not supported and won't be supported in the future.  If the
> code only supports the single threaded pool, you should probably raise
> an exception if the database isn't single threaded.
> 
> Thanks,
> Jeremy

so maybe we should keep the exception raise. What do you think, @iangreenleaf?

**Comment by iangreenleaf** [comment]
> Maybe we should add tests for the connection sharing part.

You're right, I thought I had tests that would fail without the shared connection but trying it I see they still pass. Would be great to have that tested too, but I'm happy to go ahead with this PR without that, especially given that we might not be sharing the connection for Sequel…

> Jeremy says: […] so maybe we should keep the exception raise.

Agreed that hack looks ugly and probably isn't worth it. However, I'm leaning towards the idea of making it a WARN + no-op instead. My thinking is that this gem doesn't actually *need* the connection to be shared. Generally people want it to be shared for things like transactions, but this just helps them deal with the consequences. So if someone really wants to run this with an un-shared connection, we don't actually have to stop them.

>  Maybe we should say: "Right now this gem automatically handles the following things:".

Sure, "handles" makes sense. Or "supports" might work too. I struggled on the wording of that section, since it's listing several different layers of technology 😃

**Comment by aiomaster** [comment]
I agree, so I added the warning.
The connection sharing tests are another story: #5

Thanks for maintaining that gem!

**Review by iangreenleaf — APPROVED** [review]

**Comment by iangreenleaf** [comment]
Hey, I am so sorry for abandoning this issue! Life came at me crazy and this got pushed too far down my inbox and I forgot about it. Thanks for all your work on this patch! Happy to merge it in now.

**Pull request merged**